flutter_mobx_forms/lib/src/form_field.dart

```diff
@@ -0,0 +1,115 @@
+import 'package:mobx/mobx.dart';
+
+part 'form_field.g.dart';
+
+typedef SyncFieldValidator<T> = Iterable<String> Function(T value);
+typedef AsyncFieldValidator<T> = Stream<String> Function(T value);
+
+enum ValidationPolicy { onChange, always, manual }
+
+class FormField<T> = _FormField<T> with _$FormField;
+
+abstract class _FormField<T> with Store {
```

Comment:
Flutter already use FormField as class name can we have something different to better handle friction when integration our Form with Flutter widget ?

Reply:
Suggestions? MobxFormField?

Reply:
>  MobxFormField?

I don't like it. This kind of name is acheivable using alias imports:

```dart
import 'package:flutter_mobx_form/flutter_mobx_form.dart' as mobx;

mobx.FormField foo;
```


What about:

- AutoFormField
- FieldState
- ObserverFormField

Reply:
I like `FieldState` better. It reflects the intent more clearly. Aliasing is another option but then can we use other identifiers without the alias? as in `@observable` instead of `@mobx.observabble`

Reply:
`FormField` is now `FieldState`

Reply:
Yes since `observable` is from another package

Reply:
ah nice, that's good to know. That said, I still feel `FieldState` captures the intent more clearly, we can also have `FieldGroup` for a set of related fields. Ultimately, a Form is just a `FieldGroup` at the root level. I am borrowing this concept from Angular, where I first saw it.

Reply:
Interesting so we can have reactions at a group level based on value of one of its fields or other group ?

Reply:
Yup, we can rollup errors / values all the way to the top